kinoml/modeling/homology.py

```diff
@@ -0,0 +1,272 @@
+from ..core.proteins import ProteinStructure
+from ..core.sequences import KinaseDomainAminoAcidSequence
+
+
+class HomologyModel:  #  TODO inherent a Base class?
+    def __init__(self, *args, **kwargs):
+        from appdirs import user_cache_dir
+
+        self.alipath = f"{user_cache_dir()}/alignment.ali"
+        #  TODO specify id, template, and sequence here? e.g.:
+
+        #  self.identifier = identifier
+        #  self.template = template
+        #  self.sequence = sequence
+
+    def get_pdb_template(
+        self,
+        sequence,
+    ):
+        """
+        Retrieve a template structure from PDB from a BLAST search
+        Parameters
+        ----------
+        sequence: str
+            A string of the protein sequence
+
+        Returns
+        -------
+        hits: dict
+            A dictionary generated from ProDy with PDB hits.
+        """
+
+        from prody import blastPDB
+        import tempfile
+        import pickle
+
+        blast_record = blastPDB(sequence)
+        hits = blast_record.getHits()
+
+        #  TODO need to address when BLAST search times out
+        #  TODO add option based on sequency similarity cut off
+
+        return hits
+
+    def get_sequence(
+        self, identifier: str, kinase: bool = False, backend: str = "uniprot"
+    ):
+        """
+        Retrieve a sequence based on an identifier code
+        Parameters
+        ----------
+        identifier: str
+            A string of the identifier to query (e.g 'P04629' from UniProt)
+        kinase: bool
+            Specify whether the sequence search should query kinase domains
+        backend: str
+            Specify which database to query. Options = ["uniprot", "ncbi"]
+
+        Returns
+        -------
+        sequence: str
+            A protein sequence.
+        """
+
+        import requests
+        from io import StringIO
+
+        if kinase:
+            if backend == "uniprot":
+                sequence = KinaseDomainAminoAcidSequence.from_uniprot(identifier)
+            elif backend == "ncbi":
+                sequence = KinaseDomainAminoAcidSequence.from_ncbi(identifier)
+            else:
+                raise ValueError(
+                    'Backend "%s" not supported. Please choose from ["uniprot", "ncbi"]'
+                    % (backend)
+                )
+
+        else:
+            params = {"query": identifier, "format": "fasta"}
+            response = requests.get("http://www.uniprot.org/uniprot/", params)
+
+            sequence = response.text.split("\n", 1)[1].replace("\n", "")
+
+        return sequence
+
+    def get_alignment(
```

Comment:
Check the superposition code in `opencadd` for sequence alignment: https://github.com/volkamerlab/opencadd/blob/master/opencadd/structure/superposition/sequences.py

You can use `biotite` instead, they take care of all this messy aspects of text files :D